tests/unit_tests/test_player_service.py

```diff
@@ -43,3 +43,60 @@
     player.id = -1
 
     await player_service.fetch_player_data(player)
+
+
+def test_magic_methods(player_service):
+    player = Mock()
+    player.id = 0
+    player_service[0] = player
+
+    assert len(player_service) == 1
+    assert list(iter(player_service)) == [player]
+    assert player_service[0] is player
+    assert player_service.get_player(0) is player
+
+    player_service.remove_player(player)
+
+    assert len(player_service) == 0
+    assert list(iter(player_service)) == []
+    assert player_service[0] is None
+    assert player_service.get_player(0) is None
+
+
+def test_mark_dirty(player_service):
+    player = Mock()
+    player_service[0] = player
+
+    player_service.mark_dirty(player)
+    player_service.mark_dirty(player)
```

Comment:
Why twice?

Reply:
To test uniqueness. Marking the same player as dirty multiple times should still result in the player showing up in the dirties list only once. Maybe I should add a comment

Reply:
Maybe it's pointless because the data structure is a set anyways, but I don't think it hurts to have it in there twice. You never know, maybe you get a list of sets if you mark dirty twice. It is Python after all, crazy stuff is possible.

Reply:
In that case I would put 2 asserts: one after the first mark_dirty, and one after the second.

Reply:
Yep, that's what I did